state_synchronizer/src/peer_manager.rs

```diff
@@ -200,16 +204,10 @@ impl PeerManager {
         self.requests = self.requests.split_off(&(version + 1));
     }
 
-    pub fn process_timeout(&mut self, current_requested_version: u64, timeout: u64) {
-        let request = self.requests.get(&current_requested_version).cloned();
-        if let Some((peer_id, request_time)) = request {
-            if let Some(timeout_threshold) =
-                request_time.checked_add(Duration::from_millis(timeout))
-            {
-                if SystemTime::now().duration_since(timeout_threshold).is_ok() {
-                    self.update_score(&peer_id, PeerScoreUpdateType::TimeOut);
-                    self.requests.remove(&current_requested_version);
-                }
+    pub fn process_timeout(&mut self, version: u64, penalize: bool) {
+        if let Some((peer_id, _)) = self.requests.remove(&version) {
```

Comment:
Instead of starting with self.requests.remove, shouldn't we only remove if penalize == true?